src/chip8.rs

```diff
@@ -431,32 +445,27 @@ impl Chip8 {
         let i = self.i as usize;
         let n = self.registers[register];
 
-        self.memory[i] = n / 100;
-        self.memory[i + 1] = (n / 10) % 10;
-        self.memory[i + 2] = n % 10;
+        self.memory[i + 0] = (n / 100) % 10;
+        self.memory[i + 1] = (n / 10)  % 10;
+        self.memory[i + 2] = (n / 1)   % 10;
+        self.pc += 2;
     }
 
     // Fx55 - LD [I], Vx
     // Store registers V0 through Vx in memory starting at location I.
     fn copy_registers_into_memory(&mut self, max_register: usize) {
-        for x in 0..=max_register {
+        for x in 0..max_register + 1 {
```

Comment:
🤔

Reply:
both are ugly but at least one is a widely accepted ugly since the 70s